src/COMPASS/util/runconfig.py

```diff
@@ -0,0 +1,167 @@
+from __future__ import annotations
+
+import os
+from dataclasses import dataclass
+
+import journal
+import yamale
+from ruamel.yaml import YAML
+
+from . import helpers
+
+
+def validate_group(group_cfg: dict) -> None:
+    """Check and validate runconfig entries.
+
+       Parameters
+       ----------
+       group_cfg : dict
+           Dictionary storing runconfig options to validate
+    """
+    error_channel = journal.error('runconfig.validate_group')
+
+    # Check 'input_file_group' section of runconfig
+    input_group = group_cfg['input_file_group']
+    # If is_reference flag is False, check that file path to reference
+    # burst is assigned and valid (required by geo2rdr and resample)
+    is_reference = input_group['reference_burst']['is_reference']
+    if not is_reference:
+        helpers.check_file_path(input_group['reference_burst']['file_path'])
+
+    for i, safe_file in enumerate(input_group['safe_file_path']):
+        # ensure only one safe file for reference rdr2geo processing
+        if is_reference and i > 0:
+            err_str = 'More that one safe file provided as reference'
+            error_channel.log(err_str)
+            raise ValueError(err_str)
+        helpers.check_file_path(safe_file)
+
+    helpers.check_file_path(input_group['orbit_file_path'])
+
+    # Check 'dynamic_ancillary_file_groups' section of runconfig
+    # Check that DEM file exists and is GDAL-compatible
+    dem_path = group_cfg['dynamic_ancillary_file_groups']['dem_file']
+    helpers.check_file_path(dem_path)
+    helpers.check_dem(dem_path)
+
+    # Check 'product_path_group' section of runconfig.
+    # Check that directories herein have writing permissions
+    product_path_group = group_cfg['product_path_group']
+    helpers.check_write_dir(product_path_group['product_path'])
+    helpers.check_write_dir(product_path_group['scratch_path'])
+    helpers.check_file_path(product_path_group['sas_output_file'])
+
+    # Check polarizations to process.
+    if 'polarization' not in group_cfg['processing']:
+        group_cfg['processing']['polarization'] = ['HH', 'HV', 'VH', 'VV']
```

Comment:
Note that Sentinel is not a quad-pol but a dual-pol system. May be, if no polarization is found either we read them from the SAFE file or by default we process the most used pol for OPERA (VV, VH). However, this m
![s1-constellation-mode-polarization-observation-geometry-800](https://user-images.githubusercontent.com/54076817/148320710-432486a1-252a-4a42-b92f-010ec76fadbb.png)
ay cause troubles at the North and South pol where the dual-pol mode used is HH and HV. See pic from ASF website

Reply:
How about we check the file name of the zip file and extract polarizations if not given?
The file naming convention for S1 is [here](https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/naming-conventions). For example :
S1B_IW_SLC__1S**DV**_20211127T134403_20211127T134430_029774_038DC7_08D0.zip

**DV** represents Dual Pol V, i.e., VV and VH

The different possibilities are:

SH (single HH polarisation)
SV (single VV polarisation)
DH (dual HH+HV polarisation)
DV (dual VV+VH polarisation)

Reply:
What should we do if no polarizations are provided and the polarizations in given SAFEs are inconsistent? A mix of [SH, DH] or [SV, DV] SAFEs allowed? Would there be instances near the Arctic where vertical and horizontal polarizations would be mixed?

Reply:
According to the acquisition plan here https://asf.alaska.edu/data-sets/sar-data-sets/sentinel-1/sentinel-1-observation-plans/ there might be regions in the Canadian Arctic where we might have a polarization mix up. However, they will be always belong to two different modes IW and EW (meant for sea ice)

Reply:
I'm thinking about two users:
1) our SDS:
I don't think if we should force the SDS to specify polarization. In that case then it means that we should any ways lookup the polarization and process only co-pol channel. Note that co-pol is either HH or VV. So far we have committed to process only co-pol channels for OPERA

2) community users:
Many users of coreg-slc are those that will take coreg-slc to interferometry. They usually care about co-pol channel. I think they assume the software looks up the polarization and extracts co-pol. 

NOTE:
In case if in future, OPERA will change the plan to process both polarizations, then we should change the defualt to both pols. In that case community users, should get used to providing their desired polarization.

Reply:
So to be clear, I think we "polarization" should be "optional" parameter in the run config and the default behavior should be to look up polarization and process co-pol channel.

In case if user asks for a polarization that does not exist, we should throw an exception and exit without falling back to default behavior explained above.

Reply:
@LiangJYu and @vbrancat I don't think if the current behavior of the SAS is completely consistent with what we discussed above. I wrote another comment on the [schema](https://github.com/opera-adt/COMPASS/pull/4/files#r797139128) which may clarify a bit more. Or let me know if more clarification needed or if you disagree.